Mathlib/CategoryTheory/Limits/FintypeCat.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2023 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.FintypeCat
+import Mathlib.CategoryTheory.Limits.Shapes.FiniteLimits
+import Mathlib.CategoryTheory.Limits.Types
+import Mathlib.CategoryTheory.Limits.Creates
+import Mathlib.CategoryTheory.Limits.Preserves.Finite
+import Mathlib.Data.FunLike.Fintype
+
+/-!
+# (Co)limits in the category of finite types
+
+We show that finite (co)limits exist in `FintypeCat` and that they are preserved by the natural
+inclusion `FintypeCat.incl`.
+-/
+
+open CategoryTheory Limits Functor
+
+universe u
+
+namespace CategoryTheory.Limits.FintypeCat
+
+/-- Any functor from a finite category to Types that only involves finite objects,
+has a finite limit. -/
+noncomputable def finiteLimitOfFiniteDiagram {J : Type} [SmallCategory J] [FinCategory J]
+    (K : J ⥤ Type*) (_ : (j : J) → Finite (K.obj j)) : Fintype (limit K) := by
+  have : Fintype (sections K) := Fintype.ofFinite ↑(sections K)
+  exact Fintype.ofEquiv (sections K) (Types.limitEquivSections K).symm
```

Comment:
```suggestion
instance finiteLimitOfFiniteDiagram {J : Type} [SmallCategory J] [FinCategory J]
    (K : J ⥤ Type*) [∀ (j : J), Finite (K.obj j)] : Finite (limit K) := by
  have : Fintype (sections K) := Fintype.ofFinite (sections K)
  have := Fintype.ofEquiv (sections K) (Types.limitEquivSections K).symm
  infer_instance
```

Then, you may declare an instance saying `Finite ((K ⋙ FintypeCat.incl.{u}).obj j)` which would trigger automatically in the proof of `inclusionCreatesFiniteLimits`, where you would only have to do `    have : Fintype (limit (K ⋙ FintypeCat.incl.{u})) := Fintype.ofFinite _`, and similarly for colimits.

Reply:
I followed your suggestion, this saves some repetition. I sticked with `Fintype (limit K)` instead of `Finite (limit K)` because then I don't even need the auxiliary `have : Fintype (limit (K ⋙ FintypeCat.incl.{u})) := Fintype.ofFinite _` anymore and `finiteLimitOfFiniteDiagram` is also one line shorter.